File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int count = 0;
        int sum = 0;
        int prevLength;

        while (!s.equals("1")) {
            prevLength = s.length();
            s = s.replaceAll("0", "");
            count++;
            int length = s.length();
            sum += prevLength - length;
            s = Integer.toBinaryString(length);
        }

        answer[0] = count;
        answer[1] = sum;

        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
  let answer = [0, 0];
  while (s.length > 1) {
    const newLength = s.replaceAll("0", "").length;
    answer[0]++;
    answer[1] += s.length - newLength;
    s = newLength.toString(2);
  }
  return answer;
}